contracts/Vault.vy

```diff
@@ -201,23 +205,33 @@ def close_rental(state: VaultState, sender: address) -> uint256:
     )
     payback_amount: uint256 = state.active_rental.amount - pro_rata_rental_amount
 
+    protocol_fee_amount: uint256 = pro_rata_rental_amount * state.active_rental.protocol_fee / 10000
+
     # clear active rental
     self.state = self._state_hash2(state.listing, empty(Rental))
 
     # set unclaimed rewards
-    self.unclaimed_rewards += pro_rata_rental_amount
+    self.unclaimed_rewards += pro_rata_rental_amount - protocol_fee_amount
+
+    # clear unclaimed protocol fee
+    protocol_fee_amount += self.unclaimed_protocol_fee
+    self.unclaimed_protocol_fee = 0
```

Comment:
wrt gas usage, wouldn't make sense to only do this if `self.unclaimed_protocol_fee > 0`?

Reply:
hmmm yes, you're right 👍